packages/conform-valibot/coercion.ts

```diff
@@ -0,0 +1,516 @@
+				schema: coerceArray(arraySchema),
+			};
+		}
+		case 'exact_optional': {
+			// @ts-expect-error
+			const { schema: wrapSchema } = enableTypeCoercion(type.wrapped, options);
+
+			const exactOptionalSchema = {
+				...type,
+				wrapped: wrapSchema,
+			};
+
+			return {
+				transformAction: undefined,
+				schema: exactOptionalSchema,
+			};
+		}
+		case 'nullish':
+		case 'optional': {
+			return generateWrappedSchema(type, options, true);
+		}
+		case 'undefinedable':
+		case 'nullable':
+		case 'non_optional':
+		case 'non_nullish':
+		case 'non_nullable': {
+			return generateWrappedSchema(type, options);
+		}
+		case 'union':
+		case 'intersect': {
+			const unionSchema = {
+				...type,
+				// @ts-expect-error
+				options: type.options.map(
+					// @ts-expect-error
+					(option) =>
+						enableTypeCoercion(option as GenericSchema, options).schema,
+				),
+			};
+			return {
+				transformAction: undefined,
+				schema: unionSchema,
+			};
+		}
+		case 'variant': {
+			const variantSchema = {
+				...type,
+				// @ts-expect-error
+				options: type.options.map(
+					// @ts-expect-error
+					(option) =>
+						enableTypeCoercion(option as GenericSchema, options).schema,
+				),
+			};
+			return {
+				transformAction: undefined,
+				schema: variantSchema,
+			};
+		}
+		case 'tuple': {
+			const tupleSchema = {
+				...type,
+				// @ts-expect-error
+				items: type.items.map(
+					// @ts-expect-error
+					(option) => enableTypeCoercion(option, options).schema,
+				),
+			};
+			return {
+				transformAction: undefined,
+				schema: tupleSchema,
+			};
+		}
+		case 'tuple_with_rest': {
+			const tupleWithRestSchema = {
+				...type,
+				// @ts-expect-error
+				items: type.items.map(
+					// @ts-expect-error
+					(option) => enableTypeCoercion(option, options).schema,
+				),
+				// @ts-expect-error
+				rest: enableTypeCoercion(type.rest, options).schema,
+			};
+			return {
+				transformAction: undefined,
+				schema: tupleWithRestSchema,
+			};
+		}
+		case 'loose_object':
+		case 'strict_object':
+		case 'object': {
+			const objectSchema = {
+				...type,
+				entries: Object.fromEntries(
+					// @ts-expect-error
+					Object.entries(type.entries).map(([key, def]) => [
+						key,
+						enableTypeCoercion(def as GenericSchema, options).schema,
+					]),
+				),
+			};
+
+			return {
+				transformAction: undefined,
+				schema: objectSchema,
+			};
+		}
+		case 'object_with_rest': {
+			const objectWithRestSchema = {
+				...type,
+				entries: Object.fromEntries(
+					// @ts-expect-error
+					Object.entries(type.entries).map(([key, def]) => [
+						key,
+						enableTypeCoercion(def as GenericSchema, options).schema,
+					]),
+				),
+				// @ts-expect-error
+				rest: enableTypeCoercion(type.rest, options).schema,
+			};
+
+			return {
+				transformAction: undefined,
+				schema: objectWithRestSchema,
+			};
+		}
+	}
+
+	return coerce(type, options.defaultCoercion.string);
```

Comment:
I am curious why you defaults to string coercion here as you have already handled the string case above :thinking

Reply:
This is a `switch`, that is, to make it the return value of the `enableTypeCoercion` function when it does not match the schema definition of a supported valibot.

Basically, this should be the schema definition of a supported valibot, but in case the number of schemas that can be defined in valibot increases or a schema that is not currently supported is passed, it is no exaggeration to say that this is forcibly set as the return value of the `enableTypeCoercion` function.

Reply:
If you think about it rationally, you're right, there's no need for conversion, so I've made it so that the value is returned as is.